docs/proposals/28-autonomous-shoot-clusters.md

```diff
@@ -0,0 +1,497 @@
+It downloads the required `CloudProfile`, `ControllerRegistration` and `ControllerDeployment` resources and stores them locally for usage of other `gardenadm` commands.
+This simplifies the crafting of these documents, and it also ensures that the configuration and versions used for the bootstrapping of the autonomous shoot cluster fit to the configuration later found in the connected Gardener system.
+
+#### `gardenadm init`
+
+This command bootstraps the first control plane node of the autonomous shoot cluster.
+The configuration is specified via separate files.
+It needs a `Shoot` manifest and some additional configuration files, which usually reside in the garden cluster, e.g., a `CloudProfile` (best downloaded with `gardenadm discover`).
+For the initial cluster Gardener bootstrapping use-case however, these won't come from a pre-existing garden cluster as it doesn't exist yet.
+They can then be provided locally (that's why `gardenadm discover` and `gardenadm init` are separate commands), reflecting the to-be-deployed Gardener system later.
+
+The high-level steps executed by the command are as follows:
+
+1. Generate certificate authorities, certificates and keys using the [secrets manager](../development/secrets_management.md).
+2. Generate [`OperatingSystemConfig` resource](../extensions/operatingsystemconfig.md) (in memory) containing `systemd` units and files.
+   This includes the files for the minimal standard Kubernetes control plane, i.e., `kubelet`, `etcd`, `kube-apiserver`, `kube-controller-manager`, and `kube-scheduler`.
+   Also, it should be OS-agnostic (since there is no operating system config extension running anywhere yet).
+3. Reconcile the `OperatingSystemConfig` resource using the controller logic of [`gardener-node-agent`](../concepts/node-agent.md).
+4. `kubelet` starts and runs the static pods for the minimal standard Kubernetes control plane (`etcd`, `kube-apiserver`, `kube-controller-manager` and `kube-scheduler`).
+5. Deploy [`gardener-resource-manager`](../concepts/resource-manager.md) in [host network mode](#host-network-vs-pod-network).
+6. Apply the corresponding provider and networking extensions in [host network mode](#host-network-vs-pod-network).
+7. Deploy `kube-proxy` and CoreDNS for service routing and domain name resolution.
+8. Create a [`Network` resource](../extensions/network.md) and apply it to generate `ManagedResource` associated with the pod network.
+9. Apply the corresponding provider and networking extensions in [pod network mode](#host-network-vs-pod-network).
+   Also deploy the operating system config extensions.
+10. Redeploy `gardener-resource-manager` in [pod network mode](#host-network-vs-pod-network).
+11. Create the [`OperatingSystemConfig` resource](../extensions/operatingsystemconfig.md) now in the cluster and activate the `gardener-node-agent` process via its `systemd` unit.
+12. Apply [`ControlPlane` resource](../extensions/controlplane.md).
+
+##### Problems & Solutions
```

Comment:
I see another problem: we introduced garden access for extensions so that extension controllers can easily retrieve more information about shoot clusters that is relevant to them.
This feature cannot be used when bootstrapping autonomous clusters when there is no garden cluster yet.

Right now, I have two options in mind:
- extensions are expected to correctly handle reconciliations if there simply is no garden access injected
- extensions are expected not to use information from the garden cluster if they detect autonomous clusters (e.g., because the extension resource lives in the `kube-system` namespace)

Reply:
Good point.

> extensions are expected to correctly handle reconciliations if there simply is no garden access injected

Doesn't this make sense in general? The garden access should be optional. This way, such extensions work with ASCs out of the box. When `gardenlet` gets deployed and connects to the garden, it will probably redeploy the extensions with the garden access anyways (since it takes over management of extension deployments via the `ControllerInstallation` API).

Reply:
Agreed.
However, I don't think we've implemented garden access usage in the extensions in this way. We need to document that extensions need to make this optional and skip client setup code, etc. when there is no garden access and change the implementations accordingly.